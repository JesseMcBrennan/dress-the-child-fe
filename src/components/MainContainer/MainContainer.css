@font-face {
  font-family: 'dtc-font';
  src: url("../../assets/Complete in Him.ttf"); }

.main-title {
  font-family: 'dtc-font', sans-serif;
  color: #449dba;
  font-size: 7rem;
  height: auto;
  margin: 3rem auto; }

.main-body {
  font-family: 'Source Sans Pro', sans-serif;
  font-size: 1.8rem;
  padding: 0 2rem;
  color: #615d5d;
  overflow: scroll;
  overflow-wrap: normal;
  max-height: 25rem;
  text-overflow: fade; }

.main-container {
  width: 86vw;
  height: 100vh;
  background: no-repeat center/100em url("../../images/dtc7.jpg");
  padding: 2rem 0 0 0; }

.display-card {
  height: 82vh;
  width: 62vw;
  margin: auto auto 3rem auto;
  border: solid 0.4rem whitesmoke;
  border-radius: 0.15rem;
  background-color: rgba(245, 245, 245, 0.666); }
