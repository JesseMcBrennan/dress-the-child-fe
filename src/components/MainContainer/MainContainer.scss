@font-face {
  font-family: 'dtc-font';
  src: url('../../assets/Complete in Him.ttf');
}

.main-title {
  font-family: 'dtc-font', sans-serif;
  color: #449dba;
  font-size: 6rem;
  height: auto;
  margin: 1rem auto 0.5rem auto;
}

.main-body {
  font-family: 'Source Sans Pro', sans-serif;
  font-size: 1.7rem;
  padding: 0 3rem;
  color: #615d5d;
  overflow: scroll;
  overflow-wrap: normal;
  max-height: 35rem;
}

.display-card {
  height: 82vh;
  width: 62vw;
  margin: auto;
  border: solid 0.4rem whitesmoke;
  border-radius: 0.15rem;
  background-color: rgba($color: whitesmoke, $alpha: 0.666);
}
