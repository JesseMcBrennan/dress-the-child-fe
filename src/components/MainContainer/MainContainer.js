import React, { Component } from 'react';
import { withRouter } from 'react-router-dom';
import HomePage from '../HomePage/HomePage';
import DonatePage from '../DonatePage/DonatePage';
import VolunteerPage from '../VolunteerPage/VolunteerPage';
import MissionPage from '../MissionPage/MissionPage';
import EventsPage from '../EventsPage/EventsPage';
import './MainContainer.css';

export const MainContainer = ({ location }) => {
  return (
    <section className="main-container">
      <section
        className={location.pathname === '/' ? 'none' : 'display-card'}
      >
        {location.pathname === `/` && <HomePage />}
        {location.pathname === `/donate` && <DonatePage />}
        {location.pathname === `/mission` && <MissionPage />}
        {location.pathname === `/events` && <EventsPage />}
        {location.pathname === `/volunteer` && <VolunteerPage />}
      </section>
    </section>
  );
}

export default withRouter(MainContainer);
