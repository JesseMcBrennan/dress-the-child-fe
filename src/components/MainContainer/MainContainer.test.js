import React from 'react';
import { shallow } from 'enzyme';
import { MainContainer } from './MainContainer';

describe('MainContainer', () => {
  let mockLocation;
  let wrapper;
  beforeEach(() => {
    mockLocation = {
      pathname: '/'
    };
    wrapper = shallow(<MainContainer location={mockLocation} />);
  });
  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
  it('should match the snapshot if pathname is/donate', () => {
    mockLocation = {
      pathname: '/donate'
    };
    wrapper = shallow(<MainContainer location={mockLocation} />);
    expect(wrapper).toMatchSnapshot();
  });
  it('should match the snapshot if pathname is /mission', () => {
    mockLocation = {
      pathname: '/mission'
    };
    wrapper = shallow(<MainContainer location={mockLocation} />);
    expect(wrapper).toMatchSnapshot();
  });
  it('should match the snapshot if pathname is /events', () => {
    mockLocation = {
      pathname: '/events'
    };
    wrapper = shallow(<MainContainer location={mockLocation} />);
    expect(wrapper).toMatchSnapshot();
  });
  it('should match the snapshot if pathname is /volunteer', () => {
    mockLocation = {
      pathname: '/volunteer'
    };
    wrapper = shallow(<MainContainer location={mockLocation} />);
    expect(wrapper).toMatchSnapshot();
  });
});
