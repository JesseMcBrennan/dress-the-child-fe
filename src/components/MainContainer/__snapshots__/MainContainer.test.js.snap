// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainContainer should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MainContainer
    location={
      Object {
        "pathname": "/",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <section
        className="none"
      >
        <withRouter(HomePage) />
      </section>,
      "className": "main-container",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <withRouter(HomePage) />,
          false,
          false,
          false,
          false,
        ],
        "className": "none",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        false,
        false,
        false,
        false,
      ],
      "type": "section",
    },
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <section
          className="none"
        >
          <withRouter(HomePage) />
        </section>,
        "className": "main-container",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <withRouter(HomePage) />,
            false,
            false,
            false,
            false,
          ],
          "className": "none",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          false,
          false,
          false,
          false,
        ],
        "type": "section",
      },
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`MainContainer should match the snapshot if pathname is /events 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MainContainer
    location={
      Object {
        "pathname": "/events",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <section
        className="display-card"
      >
        <EventsPage />
      </section>,
      "className": "main-container",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          false,
          false,
          false,
          <EventsPage />,
          false,
        ],
        "className": "display-card",
      },
      "ref": null,
      "rendered": Array [
        false,
        false,
        false,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        false,
      ],
      "type": "section",
    },
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <section
          className="display-card"
        >
          <EventsPage />
        </section>,
        "className": "main-container",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            false,
            false,
            false,
            <EventsPage />,
            false,
          ],
          "className": "display-card",
        },
        "ref": null,
        "rendered": Array [
          false,
          false,
          false,
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          false,
        ],
        "type": "section",
      },
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`MainContainer should match the snapshot if pathname is /mission 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MainContainer
    location={
      Object {
        "pathname": "/mission",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <section
        className="display-card"
      >
        <MissionPage />
      </section>,
      "className": "main-container",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          false,
          false,
          <MissionPage />,
          false,
          false,
        ],
        "className": "display-card",
      },
      "ref": null,
      "rendered": Array [
        false,
        false,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        false,
        false,
      ],
      "type": "section",
    },
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <section
          className="display-card"
        >
          <MissionPage />
        </section>,
        "className": "main-container",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            false,
            false,
            <MissionPage />,
            false,
            false,
          ],
          "className": "display-card",
        },
        "ref": null,
        "rendered": Array [
          false,
          false,
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          false,
          false,
        ],
        "type": "section",
      },
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`MainContainer should match the snapshot if pathname is /volunteer 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MainContainer
    location={
      Object {
        "pathname": "/volunteer",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <section
        className="display-card"
      >
        <VolunteerPage />
      </section>,
      "className": "main-container",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          false,
          false,
          false,
          false,
          <VolunteerPage />,
        ],
        "className": "display-card",
      },
      "ref": null,
      "rendered": Array [
        false,
        false,
        false,
        false,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "section",
    },
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <section
          className="display-card"
        >
          <VolunteerPage />
        </section>,
        "className": "main-container",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            false,
            false,
            false,
            false,
            <VolunteerPage />,
          ],
          "className": "display-card",
        },
        "ref": null,
        "rendered": Array [
          false,
          false,
          false,
          false,
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "section",
      },
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`MainContainer should match the snapshot if pathname is/donate 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MainContainer
    location={
      Object {
        "pathname": "/donate",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <section
        className="display-card"
      >
        <withRouter(DonatePage) />
      </section>,
      "className": "main-container",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          false,
          <withRouter(DonatePage) />,
          false,
          false,
          false,
        ],
        "className": "display-card",
      },
      "ref": null,
      "rendered": Array [
        false,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        false,
        false,
        false,
      ],
      "type": "section",
    },
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <section
          className="display-card"
        >
          <withRouter(DonatePage) />
        </section>,
        "className": "main-container",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            false,
            <withRouter(DonatePage) />,
            false,
            false,
            false,
          ],
          "className": "display-card",
        },
        "ref": null,
        "rendered": Array [
          false,
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          false,
          false,
          false,
        ],
        "type": "section",
      },
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
